core/templates/dev/head/domain/skill/SkillRightsObjectFactory.ts

```diff
@@ -17,67 +17,70 @@
  * skill rights domain objects.
  */
 
-var oppia = require('AppInit.ts').module;
-
-oppia.factory('SkillRightsObjectFactory', [
-  function() {
-    var SkillRights = function(
-        skillId, creatorId, skillIsPrivate, canEditSkillDescription) {
-      this._skillId = skillId;
-      this._creatorId = creatorId;
-      this._skillIsPrivate = skillIsPrivate;
-      this._skillDescriptionIsEditable = canEditSkillDescription;
-    };
-
-    SkillRights.prototype.getSkillId = function() {
-      return this._skillId;
-    };
-
-    SkillRights.prototype.getCreatorId = function() {
-      return this._creatorId;
-    };
-
-    SkillRights.prototype.isPrivate = function() {
-      return this._skillIsPrivate;
-    };
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
 
-    SkillRights.prototype.isPublic = function() {
-      return !this._skillIsPrivate;
-    };
+export class SkillRights {
+  _skillId: number;
```

Comment:
Actually, now that I see it, in SkillRightsObjectFactorySpec.ts, it can be both a string and a number. Hence, its type would be best suited to `number | string`

Reply:
In general, the backend and frontend should have same values type. (Make sure to validate this with the backend.)

Reply:
Done!